core/src/main/java/org/elasticsearch/index/query/PercolatorQuery.java

```diff
@@ -31,43 +30,47 @@
 import org.apache.lucene.search.Scorer;
 import org.apache.lucene.search.TwoPhaseIterator;
 import org.apache.lucene.search.Weight;
-import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.lucene.Lucene;
-import org.elasticsearch.index.fieldvisitor.SingleFieldsVisitor;
-import org.elasticsearch.index.mapper.internal.UidFieldMapper;
 import org.elasticsearch.index.percolator.ExtractQueryTermsService;
 
 import java.io.IOException;
-import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 
 import static org.apache.lucene.search.BooleanClause.Occur.FILTER;
 import static org.apache.lucene.search.BooleanClause.Occur.MUST;
 
-final class PercolatorQuery extends Query {
+public final class PercolatorQuery extends Query {
 
-    public static final float MATCH_COST =
-            (1 << 14) // stored field access cost, approximated by the number of bytes in a block
-            + 1000;   // cost of matching the query against the document, arbitrary as it would be really complex to estimate
+    // cost of matching the query against the document, arbitrary as it would be really complex to estimate
+    public static final float MATCH_COST = 1000;
 
     static class Builder {
 
+        private final String docType;
+        private final QueryRegistry queryRegistry;
+        private final BytesReference documentSource;
         private final IndexSearcher percolatorIndexSearcher;
-        private final Map<BytesRef, Query> percolatorQueries;
 
         private Query percolateQuery;
```

Comment:
this can be removed, it is not used any more. The `PercolatorService` used this if in the percolate api a `query` was specified, but since the percolator is now part of the query dsl this is redundant.